subjects/java/checkpoints/double-linkedlist/README.md

```diff
@@ -0,0 +1,144 @@
+## Double Linked List
+
+### Instructions
+
+Create a doubly linked list data structure that implements the following methods:
+
+- `at(int index)`: to access an element by its index.
+- `add(int value)`: to add an element at the end of the list.
+- `remove(int index)`: to remove an element by its index.
+- `size()`: to get the size of the list.
+
+Define these methods in an interface called `LinkedList`, and implement this interface in a class that you will create. Additionally, add private methods `next(Node node)` and `prev(Node node)` in your class that will be used in the other methods. These methods should print the messages "Go to next node" and "Go to previous node" each time they are called, respectively.
+
+### Explanation
+
+A doubly linked list is a linear data structure where each element is a separate object called a node. Each node contains three fields:
+
+- `value`: stores the data.
+- `next`: stores a reference to the next node in the list.
+- `prev`: stores a reference to the previous node in the list.
+
+The first node is called the head of the list and the last node is always null. The list allows for efficient insertion and deletion of elements. However, accessing an element by its index requires traversing the list from the head to the desired position.
```

Comment:
``` 
... and the last node is always null. ...
```

The sentence above is wrong - the last node always has `next` pointing to `null`, but the node itself shouldn't be always null

Reply:
yeah, you are right